lib/smarterEncryption.js

```diff
@@ -130,5 +130,57 @@ function generateSmarterEncryptionRuleset (domains, startingRuleId = 1) {
     return rules
 }
 
+/**
+ * Create a rule to disable or enable automatic HTTPS upgrades for a set of domains.
+ * @param {string[]} domains
+ * @param {number | undefined} id (optional) rule ID
+ * @param {'allow' | 'upgrade'} type 'allow' to create an allowlist rule, 'upgrade' to create an upgrade rule
+ * @returns {{
+ *  rule: import('./utils.js').DNRRule;
+ *  matchDetails: {
+ *      type: string;
+ *      possibleTrackerDomains?: string[];
+ *  };
+ *  }}
+ */
+function createSmarterEncryptionExceptionRule (domains, id = undefined, type = 'allow') {
+    const actionType = type === 'allow' ? 'allow' : 'upgradeScheme'
```

Comment:
Similar logic caused problem in the block list code since we assumed actions would always be block/ignore but then we added new custom rule actions. Perhaps consider only accepting the known actions and returning no rules otherwise?

Reply:
Isn't this enforced by the types? Or do you want to explicitly defend against `type` being a value beyond what the JSDoc def states?

Reply:
Well, perhaps but in practice we had very similar logic for the tracker blocking declarativeNetRequest rule generation code that became a problem. I figure it's better to avoid assuming there are only two possible values for this, but up to you.